web/profiles/joinup/src/Plugin/Block/AddContentBlock.php

```diff
@@ -21,6 +20,54 @@
 class AddContentBlock extends BlockBase {
 
   /**
+   * The collection to join.
+   *
+   * @var \Drupal\rdf_entity\RdfInterface $collection
+   */
+  protected $collection;
+
+  /**
+   * The current route match service.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface $currentRouteMatch
+   */
+  protected $currentRouteMatch;
+
+  /**
+   * The user account.
+   *
+   * @var \Drupal\user\UserInterface $account
+   */
+  protected $account;
+
+  /**
+   * The OG membership.
+   *
+   * @var \Drupal\og\Entity\OgMembership $membership
+   */
+  protected $membership;
+
+  /**
+   * Constructs a AddContentBlock object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param string $plugin_definition
+   *   The plugin implementation definition.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->currentRouteMatch = \Drupal::routeMatch();
```

Comment:
Don't inject the routematcher in a block and do direct checks on the route. That creates a hard coupling between the block and the route. If we ever change the route then this doesn't work any more.

Drupal has the concept of Contexts for this. We already have a context to retrieve the collection from the route: `\Drupal\collection\ContextProvider\CollectionRouteContext`. Let's use this instead.